src/main/java/org/mamute/controllers/TagPageController.java

```diff
@@ -47,24 +47,24 @@ public void editTagPage(String tagName, MarkedText about){
 		TagPage tagPage = tagPages.findByTag(tagName);
 		tagPage.setAbout(about);
 		if(!validator.validate(tagPage)){
-			validator.onErrorRedirectTo(TagPageController.class).editTagPageForm(tagPage.getTagName());
+			validator.onErrorRedirectTo(TagPageController.class).editTagPageForm(tagPage.getTag().getUriName());
 			return;
 		}
-		result.redirectTo(this).showTagPage(tagPage.getTagName());
+		result.redirectTo(this).showTagPage(tagPage.getTag().getUriName());
 	}
 	
 	@Post
 	@CustomBrutauthRules(ModeratorOnlyRule.class)
 	public void newTagPage(String tagName, MarkedText about){
 		if(!validator.validateCreationWithTag(tagName)) return;
-		Tag tag = tags.findByName(tagName);
+		Tag tag = tags.findBySluggedName(tagName);
 		TagPage tagPage = new TagPage(tag, about);
 		if(!validator.validate(tagPage)){
-			validator.onErrorRedirectTo(TagPageController.class).tagPageForm(tagPage.getTagName());
+			validator.onErrorRedirectTo(TagPageController.class).tagPageForm(tagPage.getTag().getUriName());
 			return;
 		}
 		tagPages.save(tagPage);
-		result.redirectTo(this).showTagPage(tagPage.getTagName()); 
+		result.redirectTo(this).showTagPage(tagPage.getTag().getUriName());
```

Comment:
Well, the problem is if `getTag` starts to receive an argument at some point, we would need to pass this new argumento at every place we call `tagPage.getTag("argument").getUriName()`.
But I understand your point, in that case we could just use a variable to keep this value so we don't need to call it multiple times :)